File: SITUBluetoothSample/ViewController.m
//
//  ViewController.m
//  SITUBluetoothSample
//
//  Created by Gary on 7/2/15.
//  Copyright (c) 2015 SITU. All rights reserved.
//

#import "ViewController.h"

#import "SITUBluetoothScaleController.h"

@interface ViewController () <SITUBluetoothDelegate>

@property (nonatomic, weak) IBOutlet UILabel *statusLabel;
@property (nonatomic, weak) IBOutlet UILabel *weight;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [self scaleDisconnected];
    
    [[SITUBluetoothScaleController sharedInstance] scanForDevice:self];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)scaleConnected
{
    [self.statusLabel setText:@"Connected"];
    [self.statusLabel setTextColor:[UIColor greenColor]];
}

- (void)scaleDisconnected
{
    [self.statusLabel setText:@"Disconnected"];
    [self.statusLabel setTextColor:[UIColor grayColor]];
}

- (void)scaleUnsupported
{
    NSLog(@"This scale is unsupported.");
}

- (void)scaleDidUpdateValue:(int)value
{
    [self.weight setText:[NSString stringWithFormat:@"%d", value]];
}

@end

File: SITUBluetoothSwiftSample/ViewController.swift
//
//  ViewController.swift
//  SITUBluetoothSwiftSample
//
//  Created by Gary on 1/25/16.
//  Copyright © 2016 SITU. All rights reserved.
//

import UIKit
import SITUBluetooth

class ViewController: UIViewController, SITUBluetoothDelegate {
    @IBOutlet weak var statusLabel : UILabel!
    @IBOutlet weak var weight : UILabel!

    override func viewDidLoad() {
        super.viewDidLoad()
        
        scaleDisconnected()
        SITUBluetoothScaleController.sharedInstance().scanForDevice(self)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    func scaleConnected()
    {
        statusLabel.text = "Connected"
        statusLabel.textColor = UIColor.greenColor()
    }
    
    func scaleDisconnected()
    {
        statusLabel.text = "Disconnected"
        statusLabel.textColor = UIColor.grayColor()
        weight.text = "N/A"
    }
    
    func scaleUnsupported()
    {
        NSLog("This scale is unsupported.");
    }
    
    func scaleDidUpdateValue(value: Int32)
    {
        weight.text = "\(value) g"
    }

}

